legal-api/src/legal_api/resources/business/business.py

```diff
@@ -41,3 +45,66 @@ def get(identifier):
             return jsonify({'message': f'{identifier} not found'}), HTTPStatus.NOT_FOUND
 
         return jsonify(business=business.json())
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def post():
+        """Create an incorporation filing, return the filing."""
+        return BusinessResource._save_incorporation_filing(request.get_json(), request)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def put(identifier):
+        """Edit an incorporation filing. REturn the filing."""
+        return BusinessResource._save_incorporation_filing(request.get_json(), request, identifier)
+
+    @staticmethod
+    def _save_incorporation_filing(incorporation_body, client_request, business_id=None):
+        """Create or update an incorporation filing. """
+        # Check that there is a JSON filing
+        if not incorporation_body:
+            return  {'message': f'No filing json data in body of post for incorporation'}, \
+                HTTPStatus.BAD_REQUEST
+        # Validate Json
+
+        temp_corp_num = incorporation_body['filing']['incorporationApplication']['nameRequest']['nrNumber']
```

Comment:
Is this nrNumber applicable for numbered company? Or this will change when we support numbered company?

Reply:
it is *likely* to change...it may be more a matter of semantics (which is why I named the variable temp corp num)